Fix: Generic parameters closing '>' isn't included in type link documentation

ericharding opened the pull request:
This fix addresses https://github.com/ionide/ionide-vscode-fsharp/issues/1164

The root cause is that html <a> tags don't nest so the end of the inside anchor tag ended the whole tag.

Before this change we would generate the following nested <a tags:
<a>TYPENAME<  <a>ARGS</a> > </a>

After this change we generate a list of <a> tags instead:
<a>TYPENAME<</a>
<a>ARGS</a>
<a> > </a>

Side note: There were a lot of functions in DocumentationFormatter.fs defined with let.. and...and...and.  I assumed this was for some legacy reason so switched them to just use `let`. I'm happy to put it back if there was a reason for it.

@baronfel @7sharp9

ericharding committed:
Remove unnecessary mutually recursive functions

ericharding committed:
Fix for ionide vscode issue 1164

baronfel commented:
Love it! I haven't yet run it locally but just reasoning through the code it seems sound. I like the change overall, thank you for sending it up.

7sharp9 committed:
Fix formatting of name where type is not generic parameter or generic argument

This fixes the issue where Thing.Thing is classed as an identifier that need envying with double ticks

Krzysztof-Cieslak commented:
Thanks a lot!
